src/main/java/de/azapps/kafkabackup/restore/message/RestoreMessageProducer.java

```diff
@@ -88,12 +90,15 @@ public void produceRecords(List<Record> recordsToProduce) {
   }
 
   private Future<RecordMetadata> produceRecord(Record record) {
-    ProducerRecord<byte[], byte[]> producerRecord = new ProducerRecord<>(record.topic(),
+    ProducerRecord<byte[], byte[]> producerRecord = new ProducerRecord(record.topic(),
         record.kafkaPartition(),
         record.timestamp(),
         record.key(),
         record.value());
 
+    record.headers().forEach(connectHeader ->
+        producerRecord.headers().add(connectHeader.key(), (byte[])connectHeader.value()));
+
```

Comment:
this won't work?
```
    ProducerRecord<byte[], byte[]> producerRecord = new ProducerRecord<>(record.topic(),
        record.kafkaPartition(),
        record.timestamp(),
        record.key(),
        record.value(),
        record.headers());
        ```

Reply:
No, we record.headers() and ProduceRecord.headers are of different types and we got cast exception